Release 2.1.2-rc1

osalyk opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/daos-stack/pmdk/17)
<!-- Reviewable:end -->

jolivier23 commented:
Testing with
https://github.com/daos-stack/daos/pull/5936
https://github.com/daos-stack/daos/pull/5793

osalyk committed:
test: call setup before require_sds

It must be called after setup because it relies
on the contents of PMEMOBJ_CONF which is set there.

osalyk committed:
test: unblock obj_sds tests for non-pmem

osalyk committed:
test: check if mock functions are used

janekmi committed:
Merge pull request #6152 from osalyk/setup_sds

test: call setup before require_sds

janekmi committed:
common: update before install (aarch64)

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
Merge pull request #6155 from pmem/main-aarch64-fix

common: update before install (aarch64)

janekmi committed:
Merge pull request #6153 from osalyk/sds_non-pmem

test: unblock obj_sds tests for non-pmem

janekmi committed:
Merge pull request #6154 from osalyk/obj_sds-test

test: check if mock functions are used

osalyk committed:
common: update workflow files

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
Merge pull request #2 from daos-stack/osalyk/workflow_files

common: update workflow files

osalyk committed:
test: add unit test for sds_check (#1)

+ minor adjustments to error/warning messages

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
common: expand the sds.at_create CTL to also cover pmemobj_open()

https://daosio.atlassian.net/browse/DAOS-17449

Ref: daos-stack/pmdk#5

Unclean cherry-pick of 0bf7f8a565aae101f17aec4bf1c168bda4fa1243

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
utils: switch self hosted runners to the new repo location
replace the outdated include command with the import_tasks command

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: switch self hosted runners to the new repo location
replace the outdated include command with the import_tasks command

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #7 from daos-stack/update_history

common: switch self hosted runners to the new repo location

osalyk committed:
common: fix typo

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #8 from daos-stack/osalyk/typo

common: fix typo

osalyk committed:
common: split the pmem_tests.yml file (#9)

Due to limited hardware availability, the previous monolithic workflow could not meet the 24-hour limit imposed by GitHub.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: reshuffle the jobs to fit both workflows into the 24-hour window

Reshuffle the jobs to fit a workflow containing the longest-running jobs into a 24-hour window.

Hence there is no longer a clear cut (with Valgrind vs without Valgrind).
They are simply called part 1 and part 2.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: extend jobs timeouts (#11)

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: add new timeout to static build

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: update the badges in the README.md file

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #12 from daos-stack/workflows_4

common: add fixes to workflow files

osalyk committed:
common: limit Static builds to ICX runners only

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
test: expand the sds.at_create CTL to also cover pmemobj_open() (fix)

- add missing log entry

Ref: daos-stack/pmdk#14

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
Merge pull request #15 from janekmi/DAOS-17449-expand_sds_at_create-master-fix

test: expand the sds.at_create CTL to also cover pmemobj_open() (fix)

osalyk committed:
common: 2.1.2-rc1 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>
